src/state.rs

```diff
@@ -3,25 +3,31 @@ use std::{convert::TryInto, sync::Arc};
 
 use crate::config::Config;
 
+/// State Builder
 #[derive(Default)]
-pub struct StateBuilder {
+pub(crate) struct StateBuilder {
```

Comment:
Why using `pub(crate)` here if the application is running in a single crate? This is the same as doing `pub` in our context. `pub(crate)` is useful when you have a library crate and want to be able to access fields as if they were public inside of the crate but developers using the crate are not able to do the same.